src/rest_framework_jwt/management/commands/obtain_token.py

```diff
@@ -0,0 +1,39 @@
+import sys
+
+from django.core.management.base import BaseCommand, CommandError
+
+
+class Command(BaseCommand):
+    help = 'Create new jwt token for the requested user'
+
+    def add_arguments(self, parser):
+        parser.add_argument('filter', type=str, help='ex: pk=7 or username=apollo or email=foo@bar')
+
+    def _parse_filter(self, filter_value):
+        sep_index = str(filter_value).find('=')
+        if sep_index == -1:
+            raise ValueError("'{}' is not a valid filter".format(filter_value))
+        lookup_field, lookup_value = filter_value[0:sep_index], filter_value[sep_index+1:]
+        return {
+            lookup_field: lookup_value
+        }
+
+    def handle(self, *args, **options):
+        try:
+            from rest_framework_jwt.settings import api_settings
+            from django.contrib.auth import get_user_model
+            user = get_user_model().objects.filter(**self._parse_filter(options['filter']))
+            if len(user) == 0:
+                raise CommandError("There is no user with the given filter")
+            if len(user) > 1:
+                raise CommandError("There are more than one user with the given filter")
+            user = user[0]
+            jwt_payload_handler = api_settings.JWT_PAYLOAD_HANDLER
+            jwt_encode_handler = api_settings.JWT_ENCODE_HANDLER
+
+            payload = jwt_payload_handler(user)
+            token = jwt_encode_handler(payload)
+            self.stdout.write(token)
+        except Exception as exp:
```

Comment:
Are `ValueError` and `CommandError` the only errors you expect here, or are there more?

Reply:
I'm afraid no, I can't predict all the possible exceptions for this block of code, and I think there isn't much I can do in `except` block other than informing the user about the error, that's why I used a catch-all `except` block.

Reply:
`except:` and `except Exception:` are two [antipatterns](https://realpython.com/the-most-diabolical-python-antipattern/) in Python. Let's say that so far, you know the only exceptions that can be raised are `ValueError` and `CommandError` and you output a nice red error message to the console that tell the user how to recover from it.

With `except Exception`, a misconfiguration error is raised that doesn't have anything to do with the library or the management command, but the output doesn't say so. Sure, it displays the error message, but it deprives you of the stack trace, meaning you have no information about the error that can help you fix it.

It's OK to use it on extremely sensitive production code that should display a user-friendly error (other than the default 500), but then you log the entire stack trace so it can be troubleshooted later.

However, this is a command-line procedure intended for developers who should be able to read stacktraces, so there is really no point in wrapping all errors, apart from the known ones.